### PR TITLE
Limit annotation to allowed namespaces

### PR DESCRIPTION
**Pull request by george-angel**

### CODE REVIEW

File: base/library/protected-annotation/src.rego [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package kubernetes.validating.annotaion
```
Comment (alkar) [inline base/library/protected-annotation/src.rego:1]: typo

Comment (alkar) [inline base/library/protected-annotation/src.rego]: typo

File: base/library/protected-annotation/src.rego [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    glob.match(input.parameters.annotationGlob, [], ns_annotations)
+    
+    # check namespace is not in the whitelist
+    not whitelisted(input.parameters.namespaceGlobWhitelist, input.review.namespace)
+
+    msg := sprintf("Namespace %v not allowed annotation %v", [input.review.namespace, ns_annotations])
+}
+
+whitelisted(namespaceGlobWhitelist, namespace) {
+    glob.match(namespaceGlobWhitelist[_], [], input.review.namespace)
```
Comment (alkar) [inline base/library/protected-annotation/src.rego]: Shouldn't the var be `namespace`?

File: base/library/protected-annotation/src.rego (hunk 1 above)
Comment (ribbybibby) [reply base/library/protected-annotation/src.rego]: I don't think it makes any practical difference, but the other policies in this repo have package names in the format:
```suggestion
package protectedannotation
```

Comment (alkar) [reply base/library/protected-annotation/src.rego]: > I don't think it makes any practical difference
Indeed, AFAIK it's only used it to share rules

File: base/library/protected-annotation/src.rego [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package kubernetes.validating.annotation
+
+violation[msg] {
```
Comment (ribbybibby) [inline base/library/protected-annotation/src.rego]:
```suggestion
violation[{"msg": msg}] {
```
I can't find this documented but I _think_ the msg field is a requirement for Gatekeeper. Certainly all the examples I can find use it.

**Review by alkar — APPROVED** [review]

File: base/library/protected-annotation/src_test.rego [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package kubernetes.validating.annotaion
```
Comment (ribbybibby) [inline base/library/protected-annotation/src_test.rego]:
```suggestion
package protectedannotation
```

The package of the tests should match the main code.

**Review by ribbybibby — APPROVED** [review]

**Pull request merged**